src/Helpers/DataResolver.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Firesphere\SolrSearch\Helpers;
+
+use Exception;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\FieldType\DBField;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\View\ArrayData;
+
+class DataResolver
+{
+    protected $component;
+    protected $columns = [];
+    protected $columnName = '';
+
+    public function __construct($component, $columns = [])
+    {
+        if (!is_array($columns)) {
+            $columns = array_filter(explode('.', $columns));
+        }
+        $this->columns = $columns;
+        $this->component = $component;
+        $this->columnName = $this->columns ? array_shift($this->columns) : null;
+    }
+
+    /**
+     * @param DataObject|ArrayData|SS_List $obj
+     * @param array|string $columns
+     *
+     * @return mixed
+     * @throws Exception
+     */
+    public static function identify($obj, $columns = [])
+    {
+        $instance = new self($obj, $columns);
+        if ($obj instanceof DataObject) {
+            return $instance->resolveDataObject();
+        }
+        if ($obj instanceof ArrayData) {
+            return $instance->resolveArrayData();
+        }
+        if ($obj instanceof SS_List) {
+            return $instance->resolveList();
+        }
+        if ($obj instanceof DBField) {
+            return $instance->resolveField();
+        }
+        throw new Exception(sprintf('Class: %s, is not supported.', get_class($obj)));
+    }
+
+    /**
+     * @param DataObject|ArrayData|SS_List $component
+     * @param array $columns
+     *
+     * @return void
+     * @throws Exception
+     */
+    protected function cannotIdentifyException($component, $columns = [])
+    {
+        if (!empty($columns)) {
```

Comment:
But what if the columns are empty?

Reply:
Good point :+1: , this is a helper method to avoid duplicate code.
Will remove the additional checks since its purpose is only to throw `Exception` nothing more and less.